rpcs3/Emu/Cell/Modules/sceNp2.cpp

```diff
@@ -750,6 +759,20 @@ error_code sceNpMatching2GetRoomSlotInfoLocal(SceNpMatching2ContextId ctxId, con
 		return SCE_NP_MATCHING2_ERROR_NOT_INITIALIZED;
 	}
 
+	roomSlotInfo->roomId             = roomId;
```

Comment:
All those infos are in the structure you receive when joining/leaving a room. You could just store it(not in ps3 memory) and change the structure here accordingly, long term that would be the best option.